nexus/authz-macros/src/lib.rs

```diff
@@ -160,8 +182,97 @@ struct Input {
+            type Value = PrimaryKeyType;
+
+            fn expecting(
+                &self,
+                formatter: &mut std::fmt::Formatter,
+            ) -> std::fmt::Result {
+                formatter.write_str(
+                    "a Rust type, or a map with a single key `uuid_kind`",
+                )
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                syn::parse_str(value)
+                    .map(PrimaryKeyType::Standard)
+                    .map_err(|e| E::custom(e.to_string()))
+            }
+
+            // seq represents a tuple type
+            fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+            where
+                A: serde::de::SeqAccess<'de2>,
+            {
+                let mut elements = vec![];
+                while let Some(element) =
+                    seq.next_element::<InputPrimaryKeyType>()?
+                {
+                    elements.push(element.0.into_external());
+                }
+                Ok(PrimaryKeyType::Standard(parse_quote!((#(#elements,)*))))
+            }
+
+            fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
+            where
+                A: serde::de::MapAccess<'de2>,
+            {
+                let key: String = map.next_key()?.ok_or_else(|| {
+                    serde::de::Error::custom("expected a single key")
+                })?;
+                if key == "uuid_kind" {
+                    // uuid kinds must be plain identifiers
+                    let value: ParseWrapper<syn::Ident> = map.next_value()?;
+                    Ok(PrimaryKeyType::new_typed_uuid(&value))
+                } else {
+                    Err(serde::de::Error::custom(
+                        "expected a single key `uuid_kind`",
+                    ))
+                }
+            }
+        }
+
+        deserializer.deserialize_any(PrimaryKeyVisitor).map(InputPrimaryKeyType)
+    }
+}
+
+// impl syn::parse::Parse for PrimaryKey {
+//     fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+//         let lookahead = input.lookahead1();
+//         // If the first element is a {, then this must be a uuid_kind.
+//         if lookahead.peek(syn::token::Brace) {
+//             let content;
+//             syn::braced!(content in input);
+//             // Once we have the content, we can parse it as a uuid_kind.
+//             let parsed = serde_tokenstream::from_tokenstream::<ParsedUuidKind>(
+//                 &content.cursor().token_stream(),
+//             )?;
+//             let kind_param = parsed.uuid_kind.into_inner();
+//             let full_ty = parse_quote!(::newtype_uuid::TypedUuid::<::omicron_uuid_kinds::#kind_param>);
+//             Ok(Self::UuidKind { full_ty })
+//         } else {
+//             Ok(Self::Standard(input.parse()?))
+//         }
+//     }
+// }
```

Comment:
Dead code?

Reply:
Yeah, thanks for catching this. I switched from using `syn::Parse` to `serde_tokenstream` but forgot to clean this up.